services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/AsyncBufferingSubscriber.java

```diff
@@ -75,7 +75,16 @@ public void onSubscribe(Subscription subscription) {
     @Override
     public void onNext(T item) {
         numRequestsInFlight.incrementAndGet();
-        CompletableFuture<?> currentRequest = consumer.apply(item);
+        CompletableFuture<?> currentRequest;
+
+        try {
+            currentRequest = consumer.apply(item);
+        } catch (Throwable t) {
+            subscription.cancel();
```

Comment:
Do we need to synchronize subscription.cancel?

Reply:
Ah good catch. Yeah look like it, I'll update.